wire/msgtx.go

```diff
@@ -215,6 +217,29 @@ func NewOutPoint(hash *chainhash.Hash, index uint32) *OutPoint {
 	}
 }
 
+// NewOutPointFromString returns a new bitcoin transaction outpoint parsed from
+// the provided string, which should be in the format "hash:index".
+func NewOutPointFromString(outpoint string) (*OutPoint, error) {
+	parts := strings.Split(outpoint, ":")
+	if len(parts) != 2 {
+		return nil, errors.New("outpoint should be of the form txid:index")
+	}
+	hash, err := chainhash.NewHashFromStr(parts[0])
+	if err != nil {
+		return nil, err
+	}
+
+	outputIndex, err := strconv.Atoi(parts[1])
```

Comment:
I think we should use `strconv.ParseInt(parts[1], 10, 64)` here, just because on a 32-bit system the plain `int` type would overflow with a full `uint32` value. It's quite unlikely for a TX to have that many outputs, I know. But I know of a specific use case where this could cause problems.